### PR TITLE
Use a single transaction to read the raw illuminance data.

### PR DESCRIPTION
**Pull request by dimpolo**
This will ensure that the data is from the same measurement

### CODE REVIEW

**Comment by acalinp** [comment]
Wow, thank you for the contribution(s)! Highly appreciated :smile: 

I think this is a more elegant read-write than what I did originally, so I will gladly merge it, however I don't think this was an issue in the first place since according to the datasheet page 19, the register values are locked until the last data register `0x8B` is read. Or maybe I'm misunderstanding the datasheet. Did you encounter an issue with this?

File: src/lib.rs [hunk 1]
```diff
@@ -234,6 +220,20 @@ where
             .map_err(Error::I2C)
             .and(Ok(data[0]))
     }
+
+    fn get_raw_data(&mut self) -> Result<RawData, Error<E>> {
+        // Read raw illuminance data
+        // Use a single transaction to ensure that the data is from the same measurement
+        // (see pg. 17 of datasheet)
```
Comment (acalinp) [inline src/lib.rs:227]: I think it's page 19, right? :smile:

Comment (dimpolo) [reply src/lib.rs:227]: image
This is the datasheet I'm reading 😅

**Comment by dimpolo** [comment]
You're welcome! Thanks for the great work! 😃 

>When the I2C read operation starts, all four ALS data registers are locked until the I2C read operation of register 0x8B is
completed. This will ensure that the data in the registers is from the same measurement even if an additional integration
cycle ends during the read operation. New measurement data is stored into temporary registers and the ALS_DATA
registers are updated as soon as there is no on-going I2C read operation.

It's kind of ambiguous... They do mention the "read operation of register 0x8B" but they also say "registers are updated as soon as there is no on-going I2C read operation".

I haven't encountered any issues yet, probably because so far I'm reading the value as soon as it's ready so a second measurement is not going to interfere.

**Pull request merged**